src/vizier/services/agent/shared/manager/registration.cc

```diff
@@ -80,6 +80,7 @@ Status RegistrationHandler::DispatchRegistration() {
   host_info->set_pod_name(agent_info()->pod_name);
   host_info->set_host_ip(agent_info()->host_ip);
   *req_info->mutable_capabilities() = agent_info()->capabilities;
+  *req_info->mutable_parameters() = agent_info()->parameters;
```

Comment:
I take it that kelvin will register itself also since it seems this manager code is shared?

Reply:
Yup, both Kelvin and PEMs use this registration code to register with the metadata service. And the metadata service distinguishes between the Kelvin/PEM using the capabilities.